utils/merge_and_push_translations.sh

```diff
@@ -54,9 +54,77 @@ fi
 
 echo "Working tree is clean, continuing"
 
+update_authors() {
+  # Update authors file
+  if python3 utils/update_authors.py; then
+    echo "Updated authors"
+  else
+    echo "Failed updating authors"
+    exit 1
+  fi
+  # Fix formatting for Lua
+  python3 utils/fix_formatting.py --lua --dir data/i18n
+  python3 utils/fix_formatting.py --lua --dir data/txts
+}
+
+update_appdata() {
+  # Update appdata
+  if python3 utils/update_appdata.py; then
+    echo "Updated appdata"
+  else
+    echo "Failed updating appdata"
+    exit 1
+  fi
+}
+
+update_statistics() {
+  # Update statistics
+  if python3 utils/update_translation_stats.py; then
+    echo "Updated translation stats"
+  else
+    echo "Failed to update translation stats"
+    exit 1
+  fi
+}
+
+gitAddGeneratedFiles() {
+  # Stage changes
+  # - Authors
+  git add 'data/txts/*.lua' || true
+  # - Locale data
+  git add 'data/i18n/*.lua' || true
+  # - Appdata
+  git add xdg/org.widelands.Widelands.appdata.xml xdg/org.widelands.Widelands.desktop || true
+  # - Statistics
+  git add data/i18n/translation_stats.conf || true
+}
+
 # Print all commands.
 set -x
 
+# Pull All translations from Transifex
+# use force to make sure really all files get pulled
+tx pull -a -f
+
+runUpdateScripts=1 # run them at least once
```

Comment:
I don't think we need this. We can probably trust git.

Reply:
I do trust git. But I do not want to run update_XXX twice when files only changed once.
What I want is:
a) translations have changed and catalogs have changed: run update_XXX for "translation commit" and again for "catalog commit"
b) only translations have changed: run update_XXX for "translation commit" only
c) only catalogs have changed: run update_XXX for "catalog commit"
d) nothing has changed: run update_XXX for (potential) "catalog commit" in case a author/... file was changed since last run of this script. If it does not update anything, no commit is created.

Reply:
Ah, I see.

(You should have included the json changes too in the descriptions. :)

So if I get it right, you want to skip `update*` on the first pull if there are no po changes, so in that case json changes will be committed on the second run, even if there are no catalog changes.

OTOH, json changes logically go together with po changes, ie. they are changes made on tx. So I think in the rare case when we get json changes, but no po changes, we should still do the updates and a "translation commit".

And regardless of the first commit, due to the extremely unlikely case when json files are edited on tx between the 2 pulls, the updates should still always be run the second time too (with `git status --short -- x.json` checks if you want to spare some cycles, see [below](https://github.com/widelands/widelands/pull/6064#discussion_r1300743261)) for consistency between the json inputs and the generated files.

Reply:
> Ah, I see.
> 
> (You should have included the json changes too in the descriptions. :)

In which description? :thinking: probably commit.
> 
> So if I get it right, you want to skip `update*` on the first pull if there are no po changes, so in that case json changes will be committed on the second run, even if there are no catalog changes.
> 

Almost (or maybe).

I want to run update_* (and do a commit) only if anything was changed by transifex (any translation in *.po or *.json). `a)` (I just double-checked I do `git status -s` without any path. So it should be for all changes by transifex.) And I want to have all resulting changes (generated) in this commit.

> OTOH, json changes logically go together with po changes, ie. they are changes made on tx. So I think in the rare case when we get json changes, but no po changes, we should still do the updates and a "translation commit".
> 
> And regardless of the first commit, due to the extremely unlikely case when json files are edited on tx between the 2 pulls, the updates should still always be run the second time too (with `git status --short -- x.json` checks if you want to spare some cycles, see [below](https://github.com/widelands/widelands/pull/6064#discussion_r1300743261)) for consistency between the json inputs and the generated files.

I want to have unexpected changes in the "translation commit" (the 2nd) `d)` 
The update_* scripts for this commit are run if they have not yet been run for "translation commit".
And they were run again when new pot files were pushed to transifex . Now this still is done, but only when really something has change. (Well, obviously something has changed. Else there would have been no push to transifex. I probably should undo this change.)

reference a) to e) reference [my comment above](#issuecomment-1685073340)

Reply:
> > (You should have included the json changes too in the descriptions. :)
> 
> In which description? 🤔 probably commit.

No, here in the comments for the cases.